iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtension.java

```diff
@@ -237,13 +239,51 @@ void updateUser(EventProcessingRequest request) throws IOException {
             addUserIdentitiesToRequest(userUpdateRequest, request);
             if (!isEmpty(userUpdateRequest.email) || !isEmpty(userUpdateRequest.userId)) {
                 userUpdateRequest.dataFields = convertAttributes(request.getUserAttributes(), shouldCoerceStrings(request));
+                if (request.getUserAttributeLists() != null && !request.getUserAttributeLists().isEmpty()) {
+                    Map<String, List<Object>> processedUserAttributeList = processUserAttributesList(request.getUserAttributeLists(), shouldCoerceStrings(request));
+                    for (Map.Entry<String, List<Object>> entry : processedUserAttributeList.entrySet()) {
+                        String key = entry.getKey();
+                        Object value = entry.getValue();
+                        userUpdateRequest.dataFields.put(key, value);
+                    }
+                }
                 Call<IterableApiResponse> preparedCall = iterableService.userUpdate(getApiKey(request), userUpdateRequest);
                 Response<IterableApiResponse> response = makeIterableRequest(preparedCall, request.getId());
                 handleIterableResponse(preparedCall, response, request.getId());
             }
         }
     }
 
+    /**
+     * Returns a type-corrected userAttributeList
+     *
+     * @param userAttributesList Map of <keys>: List<String values>
+     * @param shouldCoerceStrings boolean setting in mParticle
+     * @return a type-corrected map
+     */
+    private Map<String, List<Object>> processUserAttributesList(Map<String, List<String>> userAttributesList, boolean shouldCoerceStrings) {
+        Map<String, List<Object>> convertedAttributeList = new HashMap<>();
+        userAttributesList.forEach((key, value)-> {
+            value.forEach((attribute)-> {
+                if (!convertedAttributeList.containsKey(key)) {
+                    convertedAttributeList.put(key, new ArrayList<>());
+                }
```

Comment:
This check could be moved above the `value.forEach` call

Reply:
Also, this is a bit nitpicky, but you could avoid repeated lookups with to `convertedAttributeList.get()` in the inner loop by doing this instead:

```java
userAttributesList.forEach((key, value)-> {
    List<Object> convertedAttributes = convertedAttributeList.get(key);
    if (convertedAttributes == null) {
        convertedAttributes = new ArrayList<>();
        convertedAttributeList.put(key, convertedAttributes);
    }

    value.forEach((attribute)-> {
        if (isValidJson(attribute)) {
            // ...
            convertedAttributes.add(typeCorrectedMap); // for example
    // etc
```

Reply:
Good call, I've updated this portion to reflect your suggested optimizations